python/test/unit/language/test_core.py

```diff
@@ -3206,6 +3206,14 @@ def kernel(X, stride_xm, stride_xk, Y, stride_yk, stride_yn, W, stride_wn, strid
         z_ref = num / denom
     if epilogue == 'chain-dot':
         if 'float8' in in_dtype:
+            # Reduce z_ref's precision to fp8 to match the kernel behavior
+            if in_dtype == 'float8e4nv':
+                z_fp8 = torch.tensor(z_ref, dtype=torch.float8_e4m3fn)
+            elif in_dtype == 'float8e5':
+                z_fp8 = torch.tensor(z_ref, dtype=torch.float8_e5m2)
+            else:
+                assert "Unsupported float8 dtype"
+            z_ref = to_numpy(z_fp8.to(torch.float32))
```

Comment:
when does it cause failure? None of the existing tests are failing in CI.

Reply:
I can't verify for NVIDIA, but it failed in Intel GPU when we were using float8e4nv.
Also, we were testing a different shape: 32x32x64 instead of 128x128x64 in the original code.
The test we were running was:

`python3 -m pytest --verbose python/test/unit/language/test_core.py::test_dot[1-32-32-64-4-False-False-chain-dot-ieee-float8e4nv-float32-1]`